repoupdater.py

```diff
@@ -0,0 +1,134 @@
+import argparse
+import glob
+import os
+import re
+import sys
+import subprocess
+
+from github import Github
+from github.GithubException import GithubException
+import yaml
+
+import client
+
+ORG_NAME = "OpenSAFELY"
+REPO_NAME_PATTERN = "^.*-research$"
+BASE_PATH = os.path.abspath("research")
+
+
+def main():
+    parser = argparse.ArgumentParser(prog="repoupdater")
+    subparsers = parser.add_subparsers(help="sub-command help", dest="subcommand")
+    list_parser = subparsers.add_parser("list", help="list all repos")
+    update_parser = subparsers.add_parser(
+        "update", help="update all repos via pull or clone"
+    )
+    exec_parser = subparsers.add_parser(
+        "exec", help="execute command against all repos"
+    )
+    exec_parser.add_argument("command")
+    exec_parser.add_argument("args", nargs="*")
+    pull_request_parser = subparsers.add_parser(
+        "pull-request", help="submit pull request against each repo"
+    )
+    pull_request_parser.add_argument("branch")
+    pull_request_parser.add_argument("title")
+
+    args = parser.parse_args()
+
+    if args.subcommand == "list":
+        list_()
+    elif args.subcommand == "update":
+        update()
+    elif args.subcommand == "exec":
+        exec_([args.command] + args.args)
+    elif args.subcommand == "pull-request":
+        pull_request(args.branch, args.title)
+    else:
+        assert False, args.subcommand
+
+
+def list_():
```

Comment:
how about `list_repos()` instead of `list_()`?

Reply:
yeah, I just copied this from opensafely/repoupdater and made it use the explict config, so these are the original names. But easy enough to fix.

Reply:
ah right, do it in a follow-up PR then!